### DISCUSSION TITLE
Link Label Styling

### DISCUSSION THREAD

**Discussion by gordon-brown** [clientIO/joint#2366] [Q&A] [answered]
### Introduction

![Screenshot 2023-10-11 170638](https://github.com/clientIO/joint/assets/32515524/a55bc485-588d-4700-bec6-bb74f090897c)

A couple of questions about styling link labels that have been puzzling me. I have attached a screen shot.

1) I would like like to amend the link label (Example "Budget Exhausted") 
- Removed the surrounding shape ("stroke property?)
- Set the fill color to transparent instead of white

2) I would like to get text wrapping  to work for link labels. The text "Known Error eliminated" should ideally split over two lines because it is hiding the link underneath it.

I realize this maybe basic stuff, but I am still learning.

Thanks in advance for your help.

### Restrictions & Constraints

New User working with JointJS+. I think this is JoinJS question, but don't know enough to be sure.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
### 1. Label Styling

  ```js
  link.appendLabel({
      attrs: {
          text: {
              text: 'Label'
          },
          rect: {
              fill: 'transparent',
              stroke: 'none'
          }
      }
  }); 
  ```

  For more styling options please refer to our [tutorial](https://resources.jointjs.com/tutorial/link-labels#label-styling).

### 2. Label Text Wrap

Here's an [example](https://jsfiddle.net/kumilingus/x0q54tdL/) of how enable the automatic text wrap on labels.

**Reply by gordon-brown** [reply]
Thanks Roman.  The example was very useful. It would have been b!#^& without it.
